versioned/tiered/rocksdb/src/main/java/org/projectnessie/versioned/rocksdb/PathPattern.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.projectnessie.versioned.rocksdb;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.projectnessie.versioned.impl.condition.ExpressionPath;
+
+class PathPattern {
```

Comment:
What is this class for?

Reply:
A path is a nested structure, with each level wrapped in an optional. Parsing the path directly where it is used was getting really ugly, the deeper the path structure was. This makes it simpler to match a path against a pattern. An example would be to see if the path is fo a key in a keyMutation.

Reply:
That makes sense, there's a visitor pattern already in the Path, would it be possible to reuse that?

Reply:
This can be altered to implement ValueVisitor. I'll try that.

Reply:
Now PathPattern implements ValueVisitor and is used using the Visitor pattern.